apps/condo/domains/document/schema/Document.test.js

```diff
@@ -0,0 +1,264 @@
+                    })
+                })
+            })
+
+            it('user without employee can not', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await updateTestDocument(notEmployeeUser, document.id, {})
+                })
+            })
+
+            it('employee can not update Document to property in other organization', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await updateTestDocument(employeeUserInOtherOrganization, document.id, {})
+                })
+            })
+        })
+
+        describe('Soft delete', () => {
+            it('anonymous can not', async () => {
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await softDeleteTestDocument(anonymous, document.id)
+                })
+            })
+
+            it('admin can', async () => {
+                const [createdDocument] = await createTestDocument(admin, organization, documentCategory)
+                const [deletedDocument] = await softDeleteTestDocument(admin, createdDocument.id)
+
+                expect(deletedDocument.deletedAt).toBeDefined()
+            })
+
+            it('support can not', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await softDeleteTestDocument(support, document.id)
+                })
+            })
+
+            it('employee with canManageDocuments can', async () => {
+                const [createdDocument] = await createTestDocument(employeeUserWithDocumentPermissions, organization, documentCategory)
+                const [deletedDocument] = await softDeleteTestDocument(employeeUserWithDocumentPermissions, createdDocument.id)
+
+                expect(deletedDocument.deletedAt).toBeDefined()
+            })
+
+            it('employee without canManageDocuments can not', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await softDeleteTestDocument(employeeUserWithoutDocumentPermissions, document.id)
+                })
+            })
+
+            it('user without employee can not', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await softDeleteTestDocument(notEmployeeUser, document.id)
+                })
+            })
+
+            it('employee can not soft delete Document to property in other organization', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await softDeleteTestDocument(employeeUserInOtherOrganization, document.id)
+                })
+            })
+        })
+    })
+
+    describe('Validations', () => {
+        it('Can not create document with property organization different to document organization', async () => {
```

Comment:
Lets also check that you cannot attach a document to the deleted property or deleted organization